lib/moab/storage_repository.rb

```diff
@@ -115,19 +127,37 @@ def object_size(object_id, include_deposit=false)
       size
     end
 
+    # @param path [String] The identifier of the path to a digital object
+    # @return [Array] of hashes which contain messages for the caller
+    def verify_no_nested_moabs(path)
+      version_directories = list_sub_dirs(path).sort # sort for travis
+      results = []
+      version_directories.each do |version|
+        version_path = path + "/#{version}"
+        version_sub_dirs = list_sub_dirs(version_path).sort
+        version_sub_dir_count = version_sub_dirs.count
+        check_sub_dirs(version_sub_dir_count, version_sub_dirs, results)
+        data_dir_path = version_path + "/#{version_sub_dirs[0]}"
```

Comment:
Here, and elsewhere, we could use string interpolation, so `"#{version_path}/#{version_sub_dirs[0]}"` etc.